aip/general/0157.md

```diff
@@ -14,11 +14,35 @@ needs to give the user control over which fields it sends back.
 
 ## Guidance
 
-APIs **may** support partial responses in one of two ways (but the same API
-**must not** use both):
+APIs **may** support partial responses in one of the three ways:
+
+**Note:** If an API requires partial responses, it **should** generally decide
+a way to support across the board, but not mix and match.
+
+### Field masks parameter
+
+Field masks (`google.protobuf.FieldMask`) can be used for granting the user
+fine-grained control over what fields are returned. It is generally the
+recommended way for an API to support partial responses.
+
+To enable field masks, an
+API **should** support the masks as a [system
+parameter](https://cloud.google.com/apis/docs/system-parameters).
+This parameter can be specified either using an HTTP query parameter or an HTTP
+header.
```

Comment:
What about gRPC? Is gRPC metadata an appropriate drop in for HTTP Header?

Reply:
Yes, metadata is essentially the same as HTTP header